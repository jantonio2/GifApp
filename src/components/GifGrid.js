import React, { useEffect, useState } from 'react'

export const GifGrid = ({category}) => {

  const [count, setCount] = useState(0);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async() => {
    const url = 'https://api.giphy.com/v1/gifs/search?q=Game+of+Thrones&limit=10&api_key=G8Eh9rUQomhjc30iU4SQJ660FaIVfPgB'
    const resp = await fetch(url);
    const {data} = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      }
    })
    

    console.log(gifs);
  }

  return (
    <div>
      <h3>{category}</h3>
      <h2>{count}</h2>
      <button onClick = {() => setCount(count + 1)}></button>
    </div>
  )
}
