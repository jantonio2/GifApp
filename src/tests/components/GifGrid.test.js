import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Probando componente GifGrid', () => {
  const category = "Game of Thrones"
  
  test('debe de renderizar los componentes y hacer match con el snapshot', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow(<GifGrid category = {category}/>);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost',
      title: 'Whatever'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category = {category}/>);
    
    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
  
  
});
