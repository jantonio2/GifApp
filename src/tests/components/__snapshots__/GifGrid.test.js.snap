// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GifGrid debe de renderizar los componentes y hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Game of Thrones
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
